docs/rfcs/1012-parsec-auth.md

```diff
@@ -0,0 +1,303 @@
+
+> [!CAUTION]
+> The output of `PBKDF_A` and `PBKDF_B` should be different to avoid leaking the private key.
+
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant Alice
+  participant S as Authentification server
+
+  Alice ->> S: Provide its email for the account creation
+  S ->> Alice: Send an email with a unique code need to continue the process
+  alt Choose between password or fido2
+    Alice ->> Alice: Choose a password
+    Alice ->> Alice: Derive a secret using the password and `PBKDF_A`
+    Alice ->> Alice: Create `PRIV_AUTH_KEY`, `PUB_AUTH_KEY` using the password and `PBKDF_B`
+  else
+    Alice ->> Alice: Use the public key of the FIDO2 device as `PUB_AUTH_KEY`, the private key of FIDO2 device as `PRIV_AUTH_KEY`
+  end
+
+  Alice ->> Alice: Create the symmetric key `SYM_KEY`
+  Alice ->> Alice: Encrypt the key `SYM_KEY` with `PUB_AUTH_KEY`
+  Alice ->> S: Send the require data to finish the account creation (`enc(SYM_KEY)`, `PUB_AUTH_KEY`, `AUTH_SECRET`, email)
+```
+
+At the end of the process, the server should save the following information in a new entry:
+
+- A UUID (`user_id`)
+- The email of the user
+- The secret of the authentication method (`AUTH_SECRET`)
+
+  > [!NOTE]
+  > For password, it would be hashed using a strong hashing function.
+
+- The public key of the authentication method (`PUB_AUTH_KEY`)
+- The encrypted symmetric key (`enc(SYM_KEY)`).
+
+> [!WARNING]
+> A user should be able to register multiple authentication methods,
+> so the user entry should allow for multiple `AUTH_SECRET`, `PUB_AUTH_KEY` and `enc(SYM_KEY)`.
+
+### The authentication process
+
+To be able to authenticate the user, it will need to provide the following information:
+
+- Its email
+- The secret for the chosen authentication method (`AUTH_SECRET`)
+
+The server then verifies the provided information and return a token `AUTH_TOKEN` if the information is correct.
+That token would need to be provided to each request requiring to be authenticated.
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant Alice
+  participant S as Authentification server
+
+  Alice ->> S: Provide its email and the authentication method secret
+  S ->> S: Valide the provided information
+  alt The information are correct
+    S ->> Alice: Return an authentification token `AUTH_TOKEN`
+  else The information are not correct
+    S ->> Alice: Return a standard error for invalid authentification
+  end
+```
+
+### Uploading a new device
+
+To upload a new device, the user would need to:
+
+0. Authenticate on the server to obtain the `AUTH_TOKEN` (see previous section).
+1. Retrieve the symmetric key `enc(SYM_KEY)` from the server.
+2. Decrypt the key with `PRIV_AUTH_KEY`.
+3. Encrypt the new device with `SYM_KEY` (`ENC_DEV`).
+4. Upload the encrypted device (`ENC_DEV`) and the ID of the organization (`org_id`) to the server.
+
+The server creates an entry whose primary key is the tuple `user_id` and `org_id`.
```

Comment:
Just to be sure, it's the auth server user_id ? Not the organization user_id ?

Reply:
I've added a note to clarify that